pkg/runtime/wfengine/activity.go

```diff
@@ -189,8 +201,16 @@ loop:
 				<-t.C
 			}
 			if completed {
+				elapsed := diag.ElapsedSince(start)
+				// Activity completed, record metrics
+				diag.DefaultWorkflowMonitoring.ExecutionCount(ctx, diag.Activity, diag.StatusSuccess)
+
+				// record the time taken to complete Activity execution
+				diag.DefaultWorkflowMonitoring.ExecutionLatency(ctx, diag.Activity, diag.StatusSuccess, elapsed)
 				break loop
 			} else {
+				// Activity execution failed with recoverable error, record metrics
+				diag.DefaultWorkflowMonitoring.ExecutionCount(ctx, diag.Activity, diag.StatusRetryable)
```

Comment:
Latency should ideally be recorded even in case of failures. e.g. If an activity completed half it's execution and errored out, it should be added in latency.